bot/exts/evergreen/tic_tac_toe.py

```diff
@@ -0,0 +1,323 @@
+import asyncio
+import random
+import typing as t
+
+import discord
+from discord.ext.commands import Cog, Context, check, group, guild_only
+
+from bot.bot import Bot
+from bot.constants import Emojis
+from bot.utils.pagination import LinePaginator
+
+CONFIRMATION_MESSAGE = (
+    "{opponent}, {requester} wants to play Tic-Tac-Toe against you. React to this message with "
+    f"{Emojis.confirmation} to accept or with {Emojis.decline} to decline."
+)
+
+
+def check_win(board: t.Dict[int, str]) -> bool:
+    """Check from board, is any player won game."""
+    return any(
+        (
+            # Horizontal
+            board[1] == board[2] == board[3],
+            board[4] == board[5] == board[6],
+            board[7] == board[8] == board[9],
+            # Vertical
+            board[1] == board[4] == board[7],
+            board[2] == board[5] == board[8],
+            board[3] == board[6] == board[9],
+            # Diagonal
+            board[1] == board[5] == board[9],
+            board[3] == board[5] == board[7],
+        )
+    )
+
+
+class Player:
+    """Class that contains information about player and functions that interact with player."""
+
+    def __init__(self, user: discord.User, ctx: Context, symbol: str):
+        self.user = user
+        self.ctx = ctx
+        self.symbol = symbol
+
+    async def get_move(self, board: t.Dict[int, str], msg: discord.Message) -> t.Tuple[bool, t.Optional[int]]:
+        """
+        Get move from user.
+
+        Return is timeout reached and position of field what user will fill when timeout don't reach.
+        """
+        def check_for_move(r: discord.Reaction, u: discord.User) -> bool:
+            """Check does user who reacted is user who we want, message is board and emoji is in board values."""
+            return (
+                u.id == self.user.id
+                and msg.id == r.message.id
+                and r.emoji in board.values()
+                and r.emoji in Emojis.number_emojis.values()
+            )
+
+        try:
+            react, _ = await self.ctx.bot.wait_for('reaction_add', timeout=120.0, check=check_for_move)
```

Comment:
I think having this set to something lower like 30s for the timeout should be better as this is not a game where we would spend a lot of time thinking